crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -90,17 +90,17 @@ use std::{
 /// This will most commonly occur when working with `SystemParam`s generically, as the requirement
 /// has not been proven to the compiler.
 pub trait SystemParam: Sized {
-    type Fetch: for<'w, 's> SystemParamFetch<'w, 's>;
+    type State: SystemParamState;
 }
 
-pub type SystemParamItem<'w, 's, P> = <<P as SystemParam>::Fetch as SystemParamFetch<'w, 's>>::Item;
+pub type SystemParamItem<'w, 's, P> = <<P as SystemParam>::State as SystemParamState>::Item<'w, 's>;
 
 /// The state of a [`SystemParam`].
 ///
 /// # Safety
 ///
 /// It is the implementor's responsibility to ensure `system_meta` is populated with the _exact_
-/// [`World`] access used by the [`SystemParamState`] (and associated [`SystemParamFetch`]).
+/// [`World`] access used by the [`SystemParamState`].
 /// Additionally, it is the implementor's responsibility to ensure there is no
 /// conflicting access across all [`SystemParam`]'s.
 pub unsafe trait SystemParamState: Send + Sync + 'static {
```

Comment:
Does this and the function `get_param` have to both be marked unsafe?

Reply:
Yes. `unsafe` has different implications for fns and traits: `unsafe fn` requires the *caller* of the fn to uphold safety invariants, whereas `unsafe trait` requires the *implementer* of the trait to uphold safety invariants. In this case, we want it to be both unsafe to implement and to call, so we mark both as unsafe.

Reply:
Ah, so the other methods are unsafe to implement, but `get_param` is unsafe to call. Thanks for the clarification, I couldn't find anything online explaining why you would prefer one over the other.